### ISSUE TITLE
Refactor 1 on the Feature 8

### ISSUE THREAD

**Issue by borislav-vucicevic-04** [borislav-vucicevic-04/regressly#45]
- Refactor the utility function `gradient_descent` to calculate **mini-batch gradient descent**. 
- Implement the model `GradientDescentParams` that will hold the following:
  - weights: vector of weights
  - inputs: values of independent variables in the dataset
  - errors: list of errors
  - epochs: number of epochs
  - batch_size: size of a batch
  - learning_rate

**Comment by borislav-vucicevic-04** [comment]
Due to recent changes in plans, this feature will not be implemented
